app/code/Magento/Customer/Model/Address.php

```diff
@@ -329,6 +338,9 @@ public function afterSave()
         if ($indexer->getState()->getStatus() == StateInterface::STATUS_VALID) {
             $this->_getResource()->addCommitCallback([$this, 'reindex']);
         }
+        $quote = $this->checkoutSession->getQuote();
+        $quote->setTriggerRecollect(true);
+        $quote->getResource()->save($quote);
```

Comment:
I don't work for magento but have worked on this kind of issue before.

Would it not be safer to apply the `trigger_recollect` flag on the quote, and let the next interaction of the checkout session handled the collection and saving as part of its normal flow?